src/plumtree_broadcast.erl

```diff
@@ -374,12 +375,17 @@ code_change(_OldVsn, State, _Extra) ->
 %%%===================================================================
 handle_broadcast(false, _MessageId, _Message, Mod, _Round, Root, From, State) -> %% stale msg
     %% remove sender from eager and set as lazy
-    plumtree_util:log(debug, "moving peer ~p from eager to lazy", [From]),
+    plumtree_util:log(debug, "moving peer ~p from eager to lazy on tree rooted at ~p, requesting it to also do the same",
+                      [From, Root]),
     State1 = add_lazy(From, Root, State),
     _ = send({prune, Root, myself()}, Mod, From),
     State1;
+handle_broadcast({true, MessageId}, _OldMessageId, Message, Mod, Round, Root, From, State) ->
+    handle_broadcast(true, MessageId, Message, Mod, Round, Root, From, State);
```

Comment:
Oh, that's neat.  Can you document this in the code with a comment to explain this very case?